invenio_app_ils/circulation/receivers.py

```diff
@@ -12,18 +12,23 @@
 from invenio_circulation.signals import loan_state_changed
 
 from ..proxies import current_app_ils_extension
-from ..records.api import Item
+from ..records.api import Item, Document
 
 
 def register_circulation_signals(app):
     """Register Circulation signal."""
-    loan_state_changed.connect(index_item_after_loan_change,
+    loan_state_changed.connect(index_record_after_loan_change,
                                sender=app, weak=False)
 
 
-def index_item_after_loan_change(_, loan):
+def index_record_after_loan_change(_, loan):
     """Reindex item when attached loan changes."""
     item_pid = loan.get(Item.pid_field)
     if item_pid:
         item = Item.get_record_by_pid(item_pid)
         current_app_ils_extension.item_indexer.index(item)
+
+    document_pid = loan.get(Document.pid_field)
```

Comment:
I am not sure this is going to work. I noticed that there is no document_pid displayed on the loan list. Looked into the loans api response and the document_pid appears only inside the `document` property which gets resolved. It might is a duplication having the `document_pid` in `loan.metadata` and inside the `loan.metadata.document` but I think we need it there. I also noticed that in cli the `document_pid` was removed on the creation of loans.